evalml/automl/automl_search.py

```diff
@@ -532,7 +532,7 @@ def search(self, data_checks="auto", show_iteration_plot=True):
             X_threshold_tuning = None
             y_threshold_tuning = None
             X_train, y_train = self.X_train, self.y_train
-            if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold:
+            if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold and is_binary(self.problem_type):
```

Comment:
I just filed #1624 to track adding support here for binary timeseries pipelines. Otherwise, looks good

Reply:
Thanks for filing!